mlflow/models/signature.py

```diff
@@ -91,41 +100,55 @@ def from_dict(cls, signature_dict: Dict[str, Any]):
 
         :param signature_dict: Dictionary representation of model signature.
                                Expected dictionary format:
-                               `{'inputs': <json string>, 'outputs': <json string>" }`
+                               `{'inputs': <json string>,
+                               'outputs': <json string>,
+                               'params': <json string>" }`
 
         :return: ModelSignature populated with the data form the dictionary.
         """
         inputs = Schema.from_json(signature_dict["inputs"])
         if "outputs" in signature_dict and signature_dict["outputs"] is not None:
             outputs = Schema.from_json(signature_dict["outputs"])
-            return cls(inputs, outputs)
         else:
-            return cls(inputs)
+            outputs = None
+        if (params := signature_dict.get("params")) is not None:
+            params = ParamSchema.from_json(params)
+            return cls(inputs, outputs, params)
+        else:
+            return cls(inputs, outputs)
 
     def __eq__(self, other) -> bool:
         return (
             isinstance(other, ModelSignature)
             and self.inputs == other.inputs
             and self.outputs == other.outputs
+            and self.params == other.params
         )
 
     def __repr__(self) -> str:
         return (
             "inputs: \n"
             "  {}\n"
             "outputs: \n"
-            "  {}\n".format(repr(self.inputs), repr(self.outputs))
+            "  {}\n"
+            "params: \n"
+            "  {}\n".format(repr(self.inputs), repr(self.outputs), repr(self.params))
         )
 
 
+@experimental
```

Comment:
@serena-ruan The `params` argument should be marked experimental in the docstrig, i.e. we can just add the text `(Experimental)` to the docstring. We shouldn't mark the whole `infer_signature` method `@experimental`, since it is already a stable API; we can't make a stable API experimental, but we can add experimental arguments.

Can we make similar fixes for any other methods that we might have marked `@experimental` in this PR?